### PR TITLE
Implement IPv6 PCEP connections

### PR DESCRIPTION
**Pull request by bradyallenjohnson**

### CODE REVIEW

File: pathd/path_pcep.c [hunk 1]
```diff
@@ -248,31 +248,43 @@ DEFUN(show_pcep_counters, show_pcep_counters_cmd, "show pcep counters",
 }
 
 DEFUN_NOSH(pcep_cli_pcc, pcep_cli_pcc_cmd,
-	   "pcc [ip A.B.C.D] [port (1024-65535)] [force_stateless]",
+	   "pcc [ip A.B.C.D | ipv6 X:X::X:X] [port (1024-65535)] [force_stateless]",
```
Comment (sylane) [inline pathd/path_pcep.c:251]: The way of doing this in FRR is `ip <A.B.C.D|X:X::X:X>` see https://github.com/opensourcerouting/frr/blob/pathd-pcep/bgpd/bgp_vty.c#L5014

File: pathd/path_pcep.c [hunk 2]
```diff
@@ -314,32 +331,49 @@ DEFUN_NOSH(pcep_cli_pcc, pcep_cli_pcc_cmd,
 }
 
 DEFUN(pcep_cli_pce_opts, pcep_cli_pce_opts_cmd,
-      "pce ip A.B.C.D [port (1024-65535)] [sr-draft07]",
+      "pce [ip A.B.C.D | ipv6 X:X::X:X] [port (1024-65535)] [sr-draft07]",
```
Comment (sylane) [inline pathd/path_pcep.c:334]: Same here

File: pathd/path_pcep.c [hunk 3]
```diff
@@ -476,14 +521,20 @@ int pcep_cli_pcc_config_write(struct vty *vty)
...
-					csnprintfrr(buff, sizeof(buff),
-						    " sr-draft07");
-				vty_out(vty, " pce ip %pI4%s\n",
-					&pce_opts->addr, buff);
+				}
+				if (true == pce_opts->draft07) {
+					csnprintfrr(buff, sizeof(buff), " sr-draft07");
+				}
+				if (IS_IPADDR_V6(&pce_opts->addr)) {
+					vty_out(vty, " pce ip %pI6%s\n",
```
Comment (sylane) [inline pathd/path_pcep.c:532]: This doesn't match the command (`ipv6`), but it should be correct if the command is changed as indicated previously.

Comment (bradyallenjohnson) [reply pathd/path_pcep.c:532]: Ok, I'll fix this.

File: pathd/path_pcep.c (hunk 2 above)
Comment (bradyallenjohnson) [reply pathd/path_pcep.c:334]: Both options are used in FRR, so I'll just leave it as is.

File: pathd/path_pcep.c (hunk 1 above)
Comment (sylane) [inline pathd/path_pcep.c:251]: I think this should be `<ip A.B.C.D | ipv6 X:X::X:X>` otherwise you could specify both `ip` and `ipv6`.

**Pull request merged**